packages/lib-sourcify/test/verification.spec.ts

```diff
@@ -1,1318 +0,0 @@
-        'sources',
-        'CBORInTheMiddle',
-      );
-
-      const { contractAddress, txHash: creatorTxHash } =
-        await deployFromAbiAndBytecode(signer, contractFolderPath, []);
-
-      const checkedContracts =
-        await checkFilesWithMetadataFromContractFolder(contractFolderPath);
-      const recompiled = await checkedContracts[0].recompile();
-      const match: Match = {
-        address: contractAddress,
-        chainId: sourcifyChainHardhat.chainId.toString(),
-        runtimeMatch: null,
-        creationMatch: null,
-      };
-      const recompiledMetadata: Metadata = JSON.parse(recompiled.metadata);
-      const generateCreationCborAuxdataPositions = async () => {
-        if (!checkedContracts[0].creationBytecodeCborAuxdata) {
-          await checkedContracts[0].generateCborAuxdataPositions();
-        }
-        return checkedContracts[0].creationBytecodeCborAuxdata || {};
-      };
-      await matchWithCreationTx(
-        match,
-        recompiled.creationBytecode,
-        sourcifyChainHardhat,
-        contractAddress,
-        creatorTxHash,
-        recompiledMetadata,
-        generateCreationCborAuxdataPositions,
-        recompiled.creationLinkReferences,
-      );
-      // Must check the creationMatch
-      expect(match.creationMatch).to.equal('perfect');
-    });
-
-    it('should find a perfect match for a contract that has multiple metadata hashes in the middle of the creation bytecode', async () => {
-      const contractFolderPath = path.join(
-        __dirname,
-        'sources',
-        'CBORInTheMiddleFactory',
-      );
-      const { contractAddress, txHash: creatorTxHash } =
-        await deployFromAbiAndBytecode(signer, contractFolderPath, [
-          UNUSED_ADDRESS,
-        ]);
-      const checkedContracts =
-        await checkFilesWithMetadataFromContractFolder(contractFolderPath);
-      const recompiled = await checkedContracts[0].recompile();
-      const match: Match = {
-        address: contractAddress,
-        chainId: sourcifyChainHardhat.chainId.toString(),
-        runtimeMatch: null,
-        creationMatch: null,
-      };
-      const recompiledMetadata: Metadata = JSON.parse(recompiled.metadata);
-      const generateCreationCborAuxdataPositions = async () => {
-        if (!checkedContracts[0].creationBytecodeCborAuxdata) {
-          await checkedContracts[0].generateCborAuxdataPositions();
-        }
-        return checkedContracts[0].creationBytecodeCborAuxdata || {};
-      };
-      await matchWithCreationTx(
-        match,
-        recompiled.creationBytecode,
-        sourcifyChainHardhat,
-        contractAddress,
-        creatorTxHash,
-        recompiledMetadata,
-        generateCreationCborAuxdataPositions,
-        recompiled.creationLinkReferences,
-      );
-      expect(match.creationMatch).to.equal('perfect');
-    });
-
-    it('should not find a perfect match for a contract for which no cborAuxdataPositions could be generated', async () => {
```

Comment:
here

Reply:
we have "should return partial match when there is perfect bytecode match but no auxdata"

Reply:
I think this one is a bit different. Here we mock the `generateCreationCborAuxdataPositions` to fail. We don't have a test for this I think.

Reply:
The test you cited it's not the case it fails, it's the case in which it returns 

```
// This mocks the case where no cborAuxdataPositions could be generated
      const generateCreationCborAuxdataPositions = async () => {
        return {};
      };
      ```
      
We have that in "should handle case with no auxdata when metadata is disabled"

Reply:
true. We could add one for the case that it throws.

Reply:
I cannot make this test work with `sinon`... Do you want to try?